### PR TITLE
Add kruschke precision health indicator

### PR DESCRIPTION
**Pull request by jessie-ross**
**This PR adds the Kruschke precision health indicator.**

Part of #436 #419 and #493

See also pbmo2S-Ii-p2#comment-1788

## How has this been tested?

- Automated tests that cover added/changed functionality

### CODE REVIEW

**Comment by jessie-ross** [comment]
I would also like to add a lift ratio indicator here to detect extreme results:

```js
  // Myabe the estimate should be used instead
  const maxLiftRatio = Math.abs(analysis.metricEstimates.ratio.top < 1 ? analysis.metricEstimates.ratio.top  - 1 : analysis.metricEstimates.ratio.bottom - 1)
  indicatorDefinitions.push(
    {
      name: 'Sane lift ratio',
      value: maxLiftRatio,
      unit: HealthIndicatorUnit.Ratio,
      link: 'https://github.com/Automattic/experimentation-platform/wiki/Experiment-Health#sane-lift',
      indicationBrackets: [
        {
          max: 0.1,
          indication: {
            code: HealthIndicationCode.Nominal,
            severity: HealthIndicationSeverity.Clear,
          },
        },
        {
          max: 0.25,
          indication: {
            code: HealthIndicationCode.PossibleIssue,
            severity: HealthIndicationSeverity.Warning,
          },
        },
        {
          max: Infinity,
          indication: {
            code: HealthIndicationCode.ProbableIssue,
            severity: HealthIndicationSeverity.Error,
          },
        },
      ],
    },
  )
```

**Comment by yanirs** [comment]
> See also pbmo2S-Ii-p2#comment-1788

@jessie-ross I thought we agreed that we would _"first move things around and settle on the styling and messaging (a reference PR may work well for that), and then add more indicators"_ (end of pbmo2S-Ii-p2#comment-1791)? It's hard to see where exactly the collection of PRs is going, and we still have misleading health indicators in production.

**Comment by jessie-ross** [comment]
> @jessie-ross I thought we agreed that we would "first move things around and settle on the styling and messaging (a reference PR may work well for that), and then add more indicators" (end of pbmo2S-Ii-p2#comment-1791)? It's hard to see where exactly the collection of PRs is going, and we still have misleading health indicators in production.

I meant this as a pretty quick PR to demonstrate how the indicators would be extended. As discussed in the comments, new table version of the health report is where this is going: pbmo2S-Ii-p2#comment-1788, and will be PR'd once the refactor PR has been approved.

**Comment by yanirs** [comment]
Sounds good. Thanks for clarifying. :slightly_smiling_face:

**Comment by yanirs** [comment]
I'll hold off on reviewing until the base PR has concluded, though, as the discussion there affects this PR.

File: src/lib/analyses.ts [hunk 1]
```diff
@@ -480,3 +480,65 @@ export function getExperimentParticipantHealthIndicators(
...
 }
+
+export function getExperimentAnalysesHealthIndicators(
+  experiment: ExperimentFull,
+  analysesByStrategy: Record<AnalysisStrategy, Analysis>,
+  strategy: AnalysisStrategy,
+): HealthIndicator[] {
+  interface IndicatorDefinition extends Omit<HealthIndicator, 'indication'> {
+    indicationBrackets: Array<IndicationBracket>
+  }
```
Comment (yanirs) [inline src/lib/analyses.ts:491]: This local interface definition is unnecessary.

File: src/lib/analyses.ts [hunk 2]
```diff
@@ -480,3 +480,65 @@ export function getExperimentParticipantHealthIndicators(
...
+
+  const analysis = analysesByStrategy[strategy]
+  const metricAssignment = experiment.metricAssignments.find(
+    (metricAssignment) => metricAssignment.metricAssignmentId === analysis.metricAssignmentId,
+  )
+  // istanbul ignore next; shouldn't occur
```
Comment (yanirs) [inline src/lib/analyses.ts:497]: Let's add a test case for this and remove this comment.

File: src/lib/analyses.ts [hunk 3]
```diff
@@ -480,3 +480,65 @@ export function getExperimentParticipantHealthIndicators(
...
+  if (!metricAssignment) {
+    throw new Error('Missing metricAssignment')
+  }
+  if (!analysis.metricEstimates) {
+    return []
+  }
+
+  const indicatorDefinitions: IndicatorDefinition[] = []
```
Comment (yanirs) [inline src/lib/analyses.ts:505]: This separate declaration is unnecessary. Let's declare and initialise in one call, and let TS infer the type.

File: src/lib/analyses.ts [hunk 4]
```diff
@@ -480,3 +480,65 @@ export function getExperimentParticipantHealthIndicators(
     ...rest,
   }))
 }
+
+export function getExperimentAnalysesHealthIndicators(
```
Comment (yanirs) [inline src/lib/analyses.ts:484]: This is a bit of recurring issue... Please document exported functions as discussed in Pbmo2S-8o-p2:

> If code is meant to be used outside the module in which it’s defined, it should be fully documented (typed input arguments, typed return values, English descriptions, usage examples for non-trivial functionality, and P2 posts / FG pages as needed).

Documentation of this file needs to be improved in general as there's a lot of non-trivial stuff here, but we can start with this function.

File: src/components/experiments/single-view/results/ActualExperimentResults.tsx [hunk 5]
```diff
@@ -198,7 +198,10 @@ export default function ActualExperimentResults({
     experiment,
     primaryMetricLatestAnalysesByStrategy,
   )
-  const experimentHealthIndicators = Analyses.getExperimentParticipantHealthIndicators(experimentParticipantStats)
+  const experimentHealthIndicators = [
+    ...Analyses.getExperimentParticipantHealthIndicators(experimentParticipantStats),
+    ...Analyses.getExperimentAnalysesHealthIndicators(experiment, primaryMetricLatestAnalysesByStrategy, strategy),
```
Comment (jessie-ross) [inline src/components/experiments/single-view/results/ActualExperimentResults.tsx:203]: We keep the analyses indicators seperate so we can easily display a health report per metric assignment.

**Review by yanirs — APPROVED** [review]

File: src/lib/analyses.ts (hunk 4 above)
Comment (yanirs) [reply src/lib/analyses.ts:537]: I still think the documentation of this function is inadequate, as it requires a lot of guessing and reading the actual function to know what's going on. 

For example, it's odd that we pass in `analysesByStrategy` and `strategy` only to call `analysesByStrategy[strategy]` on the first line. Why not pass in an `analysis` then? This sort of stuff requires an explanation.

Comment (jessie-ross) [reply src/lib/analyses.ts:537]: Yeah I wanted something more like default strategy here 🤔

**Pull request merged**